File: examples/Management_11/src/urn_jaus_jss_core_Transport_1_1/Transport_ReceiveFSM.cpp


#include "urn_jaus_jss_core_Transport_1_1/Transport_ReceiveFSM.h"




using namespace JTS;

namespace urn_jaus_jss_core_Transport_1_1
{



Transport_ReceiveFSM::Transport_ReceiveFSM()
{

	/*
	 * Context has to be constructed last so that all class variables are 
	 * available if an EntryAction of the InitialState of the statemachine 
	 * needs them. 
	 */
	context = new Transport_ReceiveFSMContext(*this);

}



Transport_ReceiveFSM::~Transport_ReceiveFSM() 
{
	delete context;
}

void Transport_ReceiveFSM::setupNotifications()
{

}





};

File: examples/Java/Management_160/src/urn_jaus_jss_core_Transport_1_1/Transport_ReceiveFSM.java

package src.urn_jaus_jss_core_Transport_1_1;

import java.util.StringTokenizer;
import java.nio.ByteBuffer;
import java.nio.ByteOrder;
import framework.transport.JausRouter;
import framework.transport.JausAddress;
import framework.internalEvents.*;
import framework.StateMachine;
import framework.messages.Message;
import statemap.*;
import src.urn_jaus_jss_core_Transport_1_1.InternalEvents.*;



public class Transport_ReceiveFSM extends StateMachine{
	protected boolean running;

    Transport_ReceiveFSMContext context;

    
	
	public Transport_ReceiveFSM()
{

	/*
	 * If there are other variables, context must be constructed last so that all 
	 * class variables are available if an EntryAction of the InitialState of the 
	 * statemachine needs them. 
	 */
		context = new Transport_ReceiveFSMContext(this);

}

	
	/// Handle notifications on parent state changes
	public void setupNotifications()
	{

	}

	/// Access for debug purposes
	public String getStateName()
	{
		return context.getState().getName();
	}

	

	
}

